aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -381,6 +381,35 @@ static ScalarType get_dtype(Tensor& result, const Tensor& self, optional<ScalarT
   return src_type;
 }
 
+// Returns the appropriate floating dtype for the (floating) reduction's result
+// If the optional dtype argument is specified, that is the output dtype
+// Otherwise, the dtype of self is used, unless...
+//  (1) It is an integral type, in which case the result has the default scalar type
+static c10::optional<ScalarType> get_floating_dtype(
+    const Tensor& self,
+    const optional<ScalarType>& dtype = c10::nullopt) {
```

Comment:
I don't see any reason to make this a default.

Reply:
Death to default args.